server.js

```diff
@@ -1,30 +1,124 @@
+const mongoUrl = process.env.MONGO_URL || `mongodb+srv://${process.env.USER_ID}:${process.env.API_KEY}@cluster0.bvwog.mongodb.net/happyThoughts?retryWrites=true&w=majority`
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useCreateIndex: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 8081
 const app = express()
 
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, 'Write something, not less than 5 characters'],
+    unique: true,
+    trim: true,
+    minlength: 5, 
+    maxlength: 140,
+    // enum: ['']
+    //match: regex
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  }, 
+  createdAt: {
+    type: Date,  
+    default: Date.now
+  }
+})
+
+//Model for mongo database 
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
+//GET route
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+app.get('/thoughts', async (req, res) => {
+  // const { page, per_page } = req.query
+ 
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
+  res.json(thoughts)
+})
+
+// POST route
+app.post('/thoughts', async (req, res) => {
+  try {
+    const newThought = await new Thought ({
+      message: req.body.message}).save() 
+      res.status(201).json(newThought)
+  } catch(error) {
+    if (error.code === 11000) {
+      res.status(400).json({ error: 'Duplicated value', fields: error.keyValue })
+    }
+  res.status(400).json({ message: 'Could not send message', error})
+  }
+})
+app.post('/thoughts/:id/likes', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const updatedThought = await Thought.findOneAndUpdate(
+      { _id: id},
+      { $inc: { hearts: 1 }},
+      { new: true}
+    )
+    if (updatedThought) {
+      res.json(updatedThought)
+    } else {
+      res.status(404).json({ message: 'Not found' })
+    }
+  } catch (error) {
+    res.status(400).json({ message: 'Invalid request', error })
+  }
+})
+//DELETE route
+app.delete('thoughts/:id', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const deleteThought = await Thought.findOneAndDelete({ _id: id })
+    if (deleteThought) {
+  } else {
+    res.status(404).json({ message: 'Not found'})
+  }
+  } catch (error){
+    res.status(400).json({ message: 'Invalid request', error})
+  }
+})
+// PATCH route
+app.patch('/thoughts/:id', async (req, res) => {
+  const { id } = req.params
+
+  try{
+    const updatedThought = await Thought.findByIdAndUpdate(id, { name: req.body.name })
```

Comment:
I think here you could use (id, req.body, {new: true }). I'm a bit unsure what the part " name: req.body.name "  will do here, since it's not included in the model? There is a message, but not a name at this point. Maybe you had one and removed it?